code/modules/antagonists/chrono_dude/future_tech.dm

```diff
@@ -0,0 +1,234 @@
+ */
+
+/obj/item/implant/dust_self
+	name = "recall implant"
+	desc = "Implant that recalls an agent if the mission is concluded or things go terribly wrong."
+	icon_state = "explosive"
+	actions_types = list(/datum/action/item_action/dust_self)
+	var/popup = FALSE
+
+/datum/action/item_action/dust_self
+	check_flags = NONE
+	name = "Recall"
+
+/obj/item/implant/dust_self/on_mob_death(mob/living/L, gibbed)
+	activate("death")
+
+/obj/item/implant/dust_self/get_data()
+	var/dat = {"<b>Implant Specifications:</b><BR>
+				<b>Name:</b> FUTURE-TECH 2577 TimeSpace Agent Recall<BR>
+				<b>Life:</b> Activates upon death.<BR>
+				<b>Important Notes:</b> The target is teleported somewhere in the future.<BR>
+				<HR>
+				<b>Implant Details:</b><BR>
+				<b>Function:</b> Advanced Bluespace Technology that recalls an agent if the mission is concluded or things go terribly wrong.<BR>
+				"}
+	return dat
+
+/obj/item/implant/dust_self/activate(cause)
+	. = ..()
+	if(!cause || !imp_in)
+		return 0
+	if(cause == "action_button")
+		popup = TRUE
+		var/response = alert(imp_in, "Are you sure you want to conclude your mission? A new agent will be dispatched if your target is still alive.", "[name] Confirmation", "Yes", "No")
+		popup = FALSE
+		if(response == "No")
+			return 0
+	if(imp_in)
+		var/datum/antagonist/tca/timeagent = imp_in.mind?.has_antag_datum(/datum/antagonist/tca)
+		if (!timeagent?.mission_concluded())
+			to_chat(imp_in, "<span class='notice'>You have failed your mission. A new agent will be dispatched.</span>")
+			var/datum/antagonist/ta/prey = timeagent.prey
+			if (prey)
+				var/datum/round_event/ghost_role/chronos/hunter = new()
+				hunter.target = prey
+		else
+			to_chat(imp_in, "<span class='notice'>You mission was succesful. You will be rewarded generously.</span>")
+
+		var/turf/T = get_turf(imp_in)
+		var/obj/item/suit = imp_in.get_item_by_slot(SLOT_WEAR_SUIT)
+		if (suit?.type!=/obj/item/clothing/suit/space/chronos)
+			suit.forceMove(T)
+		var/obj/item/bag = imp_in.get_item_by_slot(SLOT_BACK)
+		if (bag?.type!=/obj/item/chrono_eraser)
+			bag.forceMove(T)
+		var/obj/item/helm = imp_in.get_item_by_slot(SLOT_HEAD)
+		if (helm?.type!=/obj/item/clothing/head/helmet/space/chronos)
+			helm.forceMove(T)
+		var/obj/item/gadget = imp_in.get_item_by_slot(SLOT_BELT)
+		if (gadget && gadget.type!=/obj/item/holosign_creator/chrono_trap && gadget.type!=/obj/item/chrono_tele && gadget.type!=/obj/item/click_remote)
+			gadget.forceMove(T)
+		var/obj/item/pocket_r = imp_in.get_item_by_slot(SLOT_R_STORE)
+		if (pocket_r && pocket_r.type!=/obj/item/holosign_creator/chrono_trap && pocket_r.type!=/obj/item/chrono_tele && pocket_r.type!=/obj/item/click_remote)
+			pocket_r.forceMove(T)
+		var/obj/item/pocket_l = imp_in.get_item_by_slot(SLOT_L_STORE)
+		if (pocket_l && pocket_l.type!=/obj/item/holosign_creator/chrono_trap && pocket_l.type!=/obj/item/chrono_tele && pocket_l.type!=/obj/item/click_remote)
+			pocket_l.forceMove(T)
```

Comment:
use istype() for type checks

Reply:
You could (and probably should) add a blacklist for the items you don't want instead of reiterating the checks over and over xd